metrics/metrics_cli.go

```diff
@@ -0,0 +1,186 @@
+package metrics
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"gopkg.in/yaml.v3"
+)
+
+type Config struct {
+	Host string `yaml:"Host"`
+	Port string `yaml:"Port"`
+
+	Type      string `yaml:"Type"`
+	Count     int    `yaml:"Count"`
+	Frequency int    `yaml:"Frequency"`
+	Random    bool   `yaml:"Random"`
+}
+
+var defaultType = "all"
+var defaultCount = 1
+var defaultFreq = 15
+var defaultRand = false
+
+type CommandLine struct{}
+
+func (c *Config) Parse(data []byte) error {
+	return yaml.Unmarshal(data, c)
+}
+
+func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "healthy")
+}
+
+// initConnection handles the metric creation and also updates the metrics via go routines
+// The delegation logic is handled here
+func (cli *CommandLine) initConnection(str string, count int, freq int, isRandom bool, conf Config) {
+
+	rand.Seed(time.Now().Unix())
+	mc := newMetricCollector()
+	mc.interval = freq
+	addressPtr := flag.String("listen_address", conf.Host+":"+conf.Port, "server listening address")
```

Comment:
```suggestion
	addressPtr := flag.String("listen_address", net.JoinHostPort(conf.Host, conf.Port), "server listening address")
```

Using [`net.JoinHostPort()`](https://golang.org/pkg/net/#JoinHostPort) should ensure that this is safe for IPv6 host address values.

Don't the flags need to be parsed after they're defined?